superset/charts/commands/importers/v1/utils.py

```diff
@@ -50,6 +50,11 @@ def import_chart(
     if '__table' in config['params']['datasource']:
         t = session.query(SqlaTable).filter(SqlaTable.uuid == config['dataset_uuid']).one()
         config['params']['datasource'] = f'{t.id}__table'
+        if qq := config.get('query_context'):
+            qq = json.loads(qq)
+            if qq.get('datasource') and qq['datasource'].get('id'):
```

Comment:
If `id` could be zero (don't think so) then check `is not None`.
And a suggestion: `qq.get('datasource', {}).get('id')`

Reply:
```
if (qq.get('datasource') or {}).get('id') is not None:
```
done, thanks